### DISCUSSION TITLE
model.Provider如何获取最新值

### DISCUSSION THREAD

**Discussion by xmsz-stu** [web-infra-dev/modern.js#6076] [Q&A]
我用api调用组件，需要用Provider包裹
但是发现组件里拿到的值是默认值而不是最新值？

然后，我想尝试useStore获取到store，然后传给Provider，发现报错` Cannot destructure property 'store' of '(0 , react__WEBPACK_IMPORTED_MODULE_1__.useContext)(...)' as it is null.`

请问如何解决？

**Comment by zllkjc** [comment]
出现错误建议提 issue，并提供下仓库这些。你这个问题看起来是 React 多实例了

**Reply by xmsz-stu** [reply]
ok  我现在先通过其他地方useStore拿到store，然后存起来，然后在传给Provider
